test/unit/enumerable_test.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'set'
+require_relative '../../lib/invoca/utils/enumerable.rb'
+require_relative '../test_helper'
+
+class EnumerableTest < Minitest::Test
+
+  context 'build_hash' do
+    should 'convert arrays of two items to hash key/value pairs' do
+      assert_equal({ 'some' => 4, 'words' => 5 }, ['some', 'short', 'words'].build_hash { |s| [s, s.length] unless s == 'short' })
+    end
+
+    should 'ignore nils returned from map' do
+      assert_equal({ 'some' => 4, 'words' => 5 }, ['some', 'short', 'words'].build_hash { |s|  s == 'short' ? nil : [s, s.length] })
+    end
+
+    # these seem like erroneous behavior, but, they have been left as-is for backward compatibility with hobosupport::Enumerable::build_hash
+
+    should 'convert arrays with one item by using the same value for key and value' do
+      assert_equal({ 'some' => 4, 'short' => 'short', 'words' => 5 }, ['some', 'short', 'words'].build_hash { |s| s == 'short' ? [s] : [s, s.length] })
+    end
+
+    should 'convert arrays with more than two items by using the last item as the value' do
+      assert_equal({ 'some' => 4, 'short' => 'three', 'words' => 5 }, ['some', 'short', 'words'].build_hash { |s| s == 'short' ? [s, 'two', 'three'] : [s, s.length] })
+    end
+
+    should 'convert empty arrays into nil keys and values' do
+      assert_equal({ 'some' => 4, nil => nil, 'words' => 5 }, ['some', 'short', 'words'].build_hash { |s| s == 'short' ? [] : [s, s.length] })
+    end
+  end
```

Comment:
These are bizarrre! I'm tempted to not port across. I seriously doubt anyone is using that weirdness.
Or a middle ground, we could use ActiveSupport::Deprecate on those. But it's probably not necessary.

Reply:
I suppose if we don't port across, those weird cases would all trigger `ArgumentError`. The only cases we need to allow--IMO--are
1. a `[key, value]` pair
2. `nil`

Reply:
Who knows if we are relying on this oddball behavior. If we are, it probably isn't covered by tests, so we are going to find out the hard way. I'd rather leave it alone.